octoprint_mrbeam/mrb_logger.py

```diff
@@ -82,7 +82,12 @@ def log(self, level, msg, *args, **kwargs):
         :param terminal_dump: Collect and log a terminal dump. Terminal dumps are also sent to analytics if analytics is not explicitly set to False.
         :type kwargs:
         """
-        msg = unicode(msg, "utf-8")
+
+        try:
+            msg = unicode(msg, "utf-8")
+        except TypeError:
+            # If it's already unicode we get this TypeError
```

Comment:
@amrsoll Axel i want you also to know this!

Reply:
I would go for
```py
if not isinstance(msg, unicode):
  msg = unicode(msg, "utf-8")
```
simpler, no comment needed

Reply:
@amrsoll:
Turns out parts of the code do pass other things than strings here (lists) and it's fine for the logger. Therefore we must keep it with the exception.
In a general level I'm with you on this philosophy to test if an action is possible before doing the action over just doing it and catching possible exceptions...